src/rime/gear/charset_filter.cc

```diff
@@ -79,9 +130,20 @@ bool CharsetFilterTranslation::LocateNextCandidate() {
 
 // CharsetFilter
 
-bool CharsetFilter::FilterText(const string& text, const string& charset) {
-  if (charset.empty()) return !contains_extended_cjk(text);
+bool CharsetFilter::FilterText(const string& text, const string& charset_with_parameter) {
+  if (charset_with_parameter.empty()) return !contains_extended_cjk(text);
+  vector<string> charset_arguments_vector;
+  boost::split(charset_arguments_vector, charset_with_parameter, boost::is_any_of("+"));
+  bool is_emoji_enabled = false;
+  if (std::find(charset_arguments_vector.begin(), charset_arguments_vector.end(), "emoji") != charset_arguments_vector.end()) {
+	is_emoji_enabled = true;
+  }
+  if (is_emoji_enabled && is_all_emoji(text)) {
+	return true;
+  }
+
   try {
+    auto charset = charset_arguments_vector[0];
```

Comment:
Sorry, my bad.
It has better be `const auto&` instead of `auto` to avoid additional copy of the string.
Though copying is not a big deal.

Reply:
Thanks for your patient code review. Learned a lot from this.